src/ui/public/fixed_scroll.js

```diff
@@ -98,11 +103,21 @@ uiModules
         listen();
       }
 
-      // reset when the width or scrollWidth of the $el changes
-      $scope.$watchMulti([
-        function () { return $el.prop('scrollWidth'); },
-        function () { return $el.width(); }
-      ], setup);
+      let width;
+      let scrollWidth;
+      function checkWidth() {
+        const newScrollWidth = $el.prop('scrollWidth');
+        const newWidth = $el.width();
+
+        if (scrollWidth !== newScrollWidth || width !== newWidth) {
+          setup();
+        }
+
+        scrollWidth = newScrollWidth;
+        width = newWidth;
```

Comment:
For slightly more performance optimization these reassignments could go into the `if` block. I'm not in favour of those state variables in general, but can't think of a better way. :wink: